### PR TITLE
Add three-phase switching function for 3-phase KKS phase-field model

### LINKED ISSUE #6857: Add switching functions for three-phase KKS phase-field model
### Description of the enhancement or error report
Add switching functions for three-phase KKS phase-field model

### Rationale for the enhancement or information for reproducing the error
Current phase-field model switching functions depend on only one order parameter. Using conventional switching functions, at a two-phase interface, there will be formation of the third phase, which is unphysical. For a three-phase system, with order parameters `eta_i`, `eta_j`, and `eta_k`, the formation of the third phase can be prevented using the switching function `h_i = eta_i^2/4 * [15 (1-eta_i) [1 + eta_i - (eta_k - eta_j)^2] + eta_i * (9eta_i^2 - 5)]` as described by Folch and Plapp, Phys. Rev. E, 72, 011602 (2005).

### Identified impact
(i.e. Internal object changes, limited interface changes, public API change, or a list of specific applications impacted)

### PR DESCRIPTION
**Pull request by laagesen**
Closes #6857 

Three-phase switching functions as described in Folch and Plapp, Phys. Rev. E, v 72, 011602 (2005). Adds tests to check for proper behavior with respect to the 3 order parameters

### CODE REVIEW

File: modules/phase_field/include/materials/SwitchingFunction3PhaseMaterial.h [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+template<>
+InputParameters validParams<SwitchingFunction3PhaseMaterial>();
+
+/**
+ * Material class to provide switching functions that prevent formation of a
+ * third phase at a two-phase interface. See Folch and Plapp, Phys. Rev. E, v. 72,
+ * 011602 (2005).
+ */
+class SwitchingFunction3PhaseMaterial : public DerivativeParsedMaterialHelper,
+                                  public ExpressionBuilder
```
Comment (dschwen) [inline modules/phase_field/include/materials/SwitchingFunction3PhaseMaterial.h]: indent

File: modules/phase_field/src/materials/SwitchingFunction3PhaseMaterial.C [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+SwitchingFunction3PhaseMaterial::SwitchingFunction3PhaseMaterial(const InputParameters & parameters) :
+    DerivativeParsedMaterialHelper(parameters),
+    _eta_i("eta_i"),
+    _eta_j("eta_j"),
+    _eta_k("eta_k")
+{
+  EBFunction h_i;
+  //Definition of the switching function for the expression builder
+  h_i(_eta_i,_eta_j,_eta_k) = _eta_i * _eta_i/4.0 * (15.0 * (1.0 - _eta_i) * (1.0 + _eta_i
+                                - (_eta_k - _eta_j) * (_eta_k - _eta_j)) + _eta_i * (9.0 * _eta_i * _eta_i - 5.0));
```
Comment (dschwen) [inline modules/phase_field/src/materials/SwitchingFunction3PhaseMaterial.C]: outdent :-)

**Comment by dschwen** [comment]
Looks good apart from the two tiny whitespace issues 👍

**Comment by permcody** [comment]
👍

**Pull request merged**